crates/red_knot_python_semantic/src/types.rs

```diff
@@ -75,13 +75,44 @@ pub(crate) fn global_symbol_ty_by_name<'db>(db: &'db dyn Db, file: File, name: &
     symbol_ty_by_name(db, global_scope(db, file), name)
 }
 
-/// Shorthand for `symbol_ty` that looks up a symbol in the builtins.
+/// Enumeration of various core stdlib modules, for which we have dedicated Salsa queries.
 ///
-/// Returns `Unbound` if the builtins module isn't available for some reason.
+/// Things will start getting very strange during type-checking if one of these doesn't exist.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub(crate) enum CoreStdlibModule {
```

Comment:
I find it surprising that this enum isn't part of `stdlib`, especially considering that they have the same variants. 

Nit: I would rename this to `Stdlib`

Reply:
> Nit: I would rename this to `Stdlib`

Well, it's not the intention that this would be an exhuastive enumeration of all stdlib modules, though -- just ones that contain core definitions for fundamental or builtin types, that we'll likely need to examine frequently during semantic analysis

Reply:
Yeah I assumed so. But it's still all the methods we need from stdlib ;) It's just that for most modules we won't have any special handling.